### ISSUE TITLE
Handle arbitrary scopes for targets

### ISSUE THREAD

**Issue by MichaelTiemannOSC** [os-climate/ITR#47]
The function _convert_projections_to_series has a named argument for scopes that defaults to S1S2.  All functions that call it currently call it with that default argument.  But this doesn't work if the scope is actually S1 or S2 or some other scope (S3, S1S2S3, potentially S1S3--which exists in the wild).  To understand how this should change, look at the larger context.  And in any case, the current scopes defined for the tool do not discuss market-based vs. location-based Scope 2 emissions, let alone the 15 categories of S3 emissions.

The function get_preprocessed_company_data (in data_warehouse.py) calls get_company_intensity_and_production_at_base_year which calls get_company_intensity_at_year which calls get_company_projected_trajectories which calls _convert_projections_to_series.  This means the principal user of _convert_projections_to_series is the calculation of cumulative emissions trajectories and targets and which need to be compared against cumulative budgets.

The global carbon budget (~ 400-450 Gt?) is the finite amount of remaining emissions the atmosphere and oceans can absorb before the climate is irreversibly ruined and all bets are off.  Scope 1 emissions are charged directly against that budget.  Scope 2 emissions are immediately traceable to power generation used in production, but emitted by others.  If we have full knowledge of all Scope 1 power generation emissions, then Scope 2 emissions should sanely fit within those.  But Scope 2 emissions should not, by themselves, be charged against the budget if the Scope 1 source of the Scope 2 emissions is known and part of the model.  The EPA says that [the 2020 update to Scope 2 guidance](https://ghgprotocol.org/scope_2_guidance) is important.

One of the benchmarks we follow is [the OECM benchmark](https://www.unepfi.org/news/themes/climate-change/new-research-identifies-remaining-global-carbon-budget-for-twelve-main-industries/), which was updated for COP26.  There's a very nice Sankey Diagram on Page 5, copied here:

![OECM-Sankey](https://user-images.githubusercontent.com/72577720/156882363-37491a3a-1032-4297-87d2-fb6c6a09ce9c.png)

TL;DR: In addition to following the overall guidance of the PAT working group, the benchmarks used by the tool must interpret and implement scoped emissions according to the definitions given in the benchmark.  In particular, for the OECM benchmark it appears that each industry is allotted their respective share of the energy generation budget as Scope 2 emissions.  The energy sector itself is NOT allocated a generous amount of S2 emissions (such as derive from power purchase agreements)--those would be double-counted.  So I think that power generation should be Scope 1 only, and everything else must have both Scope 1 and Scope 2 emissions (either as a combined metric or separately but both present).

Once the Scope 1/Scope 2 relationships, benchmarks, and reporting are sorted, the same logic could be applied to each Scope 3 category.

Trajectories are based on historic numbers, which are the total knowledge of all the scopes for the company, with each moving according to past history.  Note that companies that changing to in-sourcing or out-sourcing their power generation during the historic data period may either show unsustainable trajectories for Scope 1 vs Scope 2, or they may show precisely the trajectories that get them to their targets.

Targets can be based on specific scopes which are a subset of the total.   If we know a company has S1, S2, and S1S2 historical emissions and sets an S2 target only, do we assume S1 stays constant and that S1S2 at the target date, if S2 target is achieved, is S1 + reduced S2?  Seems reasonable and easy.

What if a company has S1 and S2 data split out but says their goal is a reduced S1+S2 total.  Do we assume they reduce in a pro-rata fashion equally?

@LeylaJavadova your thoughts?
